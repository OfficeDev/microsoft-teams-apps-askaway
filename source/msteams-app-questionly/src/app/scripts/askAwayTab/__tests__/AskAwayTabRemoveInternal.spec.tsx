/**
 * @jest-environment jsdom
 */

import * as React from 'react';
import { shallow, configure } from 'enzyme';
import enzymeToJson from 'enzyme-to-json';
import { Flex, Text, Header } from '@fluentui/react-northstar';
import Adapter from 'enzyme-adapter-react-16';
import AskAwayTabRemoveInternal from '../AskAwayTabRemoveInternal';

configure({ adapter: new Adapter() });
jest.mock('react-i18next', () => ({
    useTranslation: () => {
        return {
            t: (str) => str,
        };
    },
}));

describe('AskAwayTabRemoveInternal Component', () => {
    afterAll(() => {
        jest.resetAllMocks();
    });

    it('should match the snapshot', () => {
        const wrapper = shallow(<AskAwayTabRemoveInternal />);

        expect(enzymeToJson(wrapper)).toMatchSnapshot();
    });

    it('should render the tab', () => {
        const component = shallow(<AskAwayTabRemoveInternal />);

        expect(component.find(Flex)).toHaveLength(1);
        expect(component.find(Text)).toHaveLength(1);
        expect(component.find(Header)).toHaveLength(1);
    });
});
