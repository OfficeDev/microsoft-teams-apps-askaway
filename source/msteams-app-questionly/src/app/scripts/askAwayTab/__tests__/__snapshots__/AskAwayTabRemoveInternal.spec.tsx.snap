// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AskAwayTabRemoveInternal Component should match the snapshot 1`] = `
<Flex
  fill={true}
>
  <FlexItem>
    <div>
      <Header
        as="h1"
        content="tab.removeHeaderText"
      />
      <Text
        as="span"
        content="tab.removeTextNotification"
      />
    </div>
  </FlexItem>
</Flex>
`;
